### PR TITLE
Account for existing _forceLastValidStreamState when rebuilding

### PR DESCRIPTION
**Pull request by kuhnroyal**
* potential fix for #89

### CODE REVIEW

**Comment by kuhnroyal** [comment]
There are 2 test failures in the persistence test but that has to be unrelated.

**Comment by marcglasberg** [comment]
Yes, the persistence code is correct, but the persistence tests that are failing because they depend on precise timing. I was hoping to think of a way to make them more stable, but maybe they need to be removed.

**Review by marcglasberg — APPROVED** [review]

**Comment by marcglasberg** [comment]
Ideally we should first create a failing test that was fixed by this change.

**Comment by kuhnroyal** [comment]
> Ideally we should first create a failing test that was fixed by this change.

I agree but that needs to be a widget test. Will see what I can do.

**Comment by marcglasberg** [comment]
Yes, if the problem is in the `didUpdateWidget` you need a widget test. Thank you!!

**Comment by kuhnroyal** [comment]
I added a test case which fails without the change.

I also needed to make one more change in the `StoreConnector`. Without setting the initial state as `_mostRecentValidState`, it could still fail  if `shouldUpdateModel` returns `false` for the first state change.
In that case `_mostRecentValidState` would still be `null` in line https://github.com/marcglasberg/async_redux/blob/master/lib/src/store_connector.dart#L477

**Comment by kuhnroyal** [comment]
I added a test workflow, if you want to enable Github Actions.

**Comment by kuhnroyal** [comment]
Additionally I found with my test that the `converter` does not work together with `shouldUpdateModel`.
The incorrect state is passed to the converter function in https://github.com/marcglasberg/async_redux/blob/master/lib/src/store_connector.dart#L600

**Comment by kuhnroyal** [comment]
We could an an assert or change the converter function parameter, both changes are breaking.

**Comment by marcglasberg** [comment]
What changes to the converter function do you recommend?

File: lib/src/store_connector.dart [hunk 1]
```diff
@@ -378,6 +378,12 @@ class _StoreStreamListenerState<St, Model> //
     }
 
     _computeLatestModel();
+    if (widget.shouldUpdateModel != null) {
+      // The initial state has to be valid at this point.
+      // This is need so that the first stream event
```
Comment (marcglasberg) [inline lib/src/store_connector.dart:383]: Type here. "This is needed"

**Comment by kuhnroyal** [comment]
Fixed the typo.

> What changes to the converter function do you recommend?

I would say, release the fix as a patch and change the parameter of the `converter` function in a breaking release.
So far nobody seems to have this problem, I hardly ever use the converter and only noticed this with the test case.

**Comment by marcglasberg** [comment]
@kuhnroyal Your fix is published in **version 17.0.1**.

**Comment by marcglasberg** [comment]
@kuhnroyal Your fix is published in version 17.0.1. Thank you!

**Pull request merged**